include/deal.II/differentiation/ad/sacado_math.h

```diff
@@ -0,0 +1,96 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_differentiation_ad_sacado_math_h
+#define dealii_differentiation_ad_sacado_math_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_TRILINOS
+
+#include <deal.II/base/numbers.h>
+#include <deal.II/differentiation/ad/ad_number_traits.h>
+#include <deal.II/differentiation/ad/sacado_number_types.h>
+
+
+#ifndef DOXYGEN
+
+/**
+ * Define some missing fundamental math functions
+ */
+namespace std
+{
+
+  /**
+   * Implementation of the error function for real-valued Sacado numbers.
+   */
+  template<typename ADNumberType, typename = typename std::enable_if<
+             dealii::Differentiation::AD::is_sacado_number<ADNumberType>::value &&
+             dealii::Differentiation::AD::is_real_valued_ad_number<ADNumberType>::value
+             >::type>
+  inline ADNumberType erf(ADNumberType x)
+  {
+    // Reference:
+    // Handbook of Mathematical Functions: with Formulas, Graphs, and Mathematical Tables
+    // Abramowitz, M. and Stegun, I.
+    // Dover Books on Mathematics. 1972.
```

Comment:
that book is older than any of us -- and it's still useful :-)

Reply:
What? You were born after 1972? ;-P

Reply:
Yes, I'm that young. Only @guidokanschat is older than the book :-)

I was in a bar yesterday and had my ID card checked. We then had a conversation about who gets carded and the waitress said that the rule is that everyone who looks younger than 40 needs to be checked, so that they err on the side of safety. I took it as a compliment that she thought she needed to card me ;-)

Reply:
That's awesome :-) With that reasoning, who could get annoyed at having your ID checked... My first experience with this in the US was in an Indian restaurant. A young (maybe 12 years old?) kid was helping his parents by manning the cash register. When I took a beer to accompany my buffet dinner, the youngster requested to see my ID. I thought that was quite funny.